de.marw.cmake.cdt/src/main/java/de/marw/cmake/cdt/internal/lsp/ParserDetection.java

```diff
@@ -64,7 +64,8 @@ private static void init() {
       final IArglet[] gcc_args = { new Arglets.IncludePath_C_POSIX(), new Arglets.MacroDefine_C_POSIX(),
           new Arglets.MacroUndefine_C_POSIX(),
           // not defined by POSIX, but does not harm..
-          new Arglets.SystemIncludePath_C(), new Arglets.LangStd_GCC(), new Arglets.Sysroot_GCC(),
+          new Arglets.SystemIncludePath_C(), new Arglets.LangStd_GCC(), 
+          new Arglets.Sysroot_GCC(), new Arglets.Target_Clang(), 
```

Comment:
Target_Clang() is an option gcc does not understand.
Starting with your contribution, the parsers for gcc and clang begin to derivate.
For maintenance purposes, it would be cleaner here to have separate code sections for the two compilers.
Could you move the clang support to sections of its own (clang C and clang  C++)
similar to what is in line #87++?

Reply:
Since all the gcc_args are also clang args, I would rename the gcc_args array to common_args, and add a new clang_args array (and eventually a gcc_args array) and create a copy to pass to the CommandlineParser (So a new list containing the previous array plus clang specific ones,and then call toarray on it) is that OK?

Reply:
Please stay with gcc_args array for gcc and introduce a new clang_args array of its own.
That static initalization currently is way too complicated, so pls do not put in extra comuptaions.